RTP Event DTMF not properly recognized in Version 1.1.0

Baresip is "50" and calls "23" in below log. Versions are re/v2.0.1, rem/v1.0.0, baresip/v1.1.0. The receiver, "23", accepts the call and types "1" on its keypad, which is recognized by baresip - see "received in-band DTMF event: '1' (end=0)" below. **The problem is, "received in-band DTMF event: '1' (end=1)" is missing** and instead, it writes something about discarding old audio frames. When i switch back to Version baresip/v1.0.0 (same re/rem as above) - it works! "received event: '1' (end=0)" AND  "received event: '1' (end=1)" is coming, and no note about discarding audio frames.

	  audio: update
	  audio: Set audio decoder: PCMA 8000Hz 1ch
	  audio: start
	  audio: create recv buffer [20 - 160 ms] [320 - 2560 bytes]
	  alsa: reset: srate=8000, ch=1, num_frames=160, pcmfmt=S16_LE
	  alsa: playback started (default)
	  audio: player started with sample format S16LE
	  audio: Set audio encoder: PCMA 8000Hz 1ch
	  audio: start
	  aubridge: created device 'nil2'
	  audio: source started with sample format S16LE
	  audio tx pipeline:    aubridge ---> PCMA
	  audio rx pipeline:        alsa <--- PCMA
	  call: stream start (active=1)
	  audio: start
	  50@192.168.20.11: Call established: sip:23@192.168.20.11
	  ctrl_tcp: handle_command:  cmd='ausrc', params:'gst,file:///var/www/html/cache/dialer-standalone/alarm.mp3', token=''
	  stream: incoming rtp for 'audio' established, receiving from 192.168.20.11:15288
	  gst: title: Alarm
	  gst: title: Alarm
	  gst: title: Alarm/0 (bit/s)    
	  gst: title: Alarm/0 (bit/s)    
	  ctrl_tcp: handle_command:  cmd='ausrc', params:'aubridge,nil2', token=''
	  aubridge: created device 'nil2'
	  #################### I PRESS "1" KEY ON PAD
	  received in-band DTMF event: '1' (end=0)
	  audio: [time=27.699] discard old frame (0.000 seconds old)
	  audio: [time=27.699] discard old frame (0.000 seconds old)
	  audio: [time=27.699] discard old frame (0.000 seconds old)
	  #################### AGAIN, I PRESS "1" KEY ON PAD - "received in-band DTMF event" IS MISSING BUT SOMETHING IS, AGAIN, DISCARDED. I AM GUESSING THIS IS THE end=1 EVENT.
	  audio: [time=40.590] discard old frame (0.000 seconds old)
	  audio: [time=40.590] discard old frame (0.000 seconds old)
	  audio: [time=40.590] discard old frame (0.000 seconds old)
	  #################### I PRESS "2" KEY ON PAD
	  received in-band DTMF event: '2' (end=0)
	  50@192.168.20.11: selected for 50-0x875bd0
	  ua: incoming OPTIONS message from sip:asterisk@192.168.20.11 (192.168.20.11:5060)
	  ua: using best effort AF: af=AF_INET
	  call: alloc with params laddr=192.168.20.237, af=AF_INET, use_rtp=0
	  call: use_video=0
	  audio: destroyed (started=0)
	  audio: [time=49.200] discard old frame (0.000 seconds old)
	  audio: [time=49.200] discard old frame (0.000 seconds old)
	  audio: [time=49.200] discard old frame (0.000 seconds old)

Wireshark looks like this
![rtp-event](https://user-images.githubusercontent.com/13996968/122038772-3e7ce500-cdd6-11eb-9093-3dafcf6bcb8d.jpg)

So it has to be an issue with this diff (v1.0.0 to v1.1.0) - https://github.com/baresip/baresip/compare/1ef6826..82123b6 - but that is very.... long O_O

I have tried to compile HEAD (with re/v2.0.1, rem/v1.0.0 still installed). It throws many compiler errors about forward declarations. So I compiled those first in their HEAD state, also:

	  rm -rf /usr/src/re* /usr/local/include/re* /usr/local/lib/libre*;cd /usr/src;ldconfig
	  cd /usr/src
	  git clone --depth 1 https://github.com/baresip/re
	  cd re
	  ldconfig;make clean;make install;ldconfig
	  cd ..
	  git clone --depth 1 https://github.com/baresip/rem
	  cd rem
	  ldconfig;make clean;make install;ldconfig
	  cd ..
	  rm -rf /usr/src/baresip* /usr/local/bin/baresip /usr/local/lib/baresip/modules/*;cd /usr/src;ldconfig
	  git clone --depth 1 https://github.com/baresip/baresip
	  cd baresip
	  ldconfig;make clean;make install;ldconfig
	  
	  Fails with:
	  
	  ..
	    CC [M]  build-x86_64/modules/g711/g711.o
	    LD [M]  g711.so
	    CC [M]  build-x86_64/modules/gsm/gsm.o
	    LD [M]  gsm.so
	    GEN     modules/ctrl_dbus/com.github.Baresip.xml
	  Usage: gdbus-codegen [options]
	  
	  gdbus-codegen: error: no such option: --output-directory
	  modules/ctrl_dbus/module.mk:26: recipe for target 'modules/ctrl_dbus/baresipbus.h' failed
	  make: *** [modules/ctrl_dbus/baresipbus.h] Error 2

The same using --branch v2.0.1, --branch v1.0.0 and --branch v1.1.0 as git command line parameters on the re, rem and baresip respectively, works.

There should be a compatibility table which version of re/rem works with which version of baresip.

--- a/src/core.h
+++ b/src/core.h
@@ -286,7 +286,6 @@ struct stream {
 	uint32_t pseq;           /**< Sequence number for incoming RTP      */
 	bool pseq_set;           /**< True if sequence number is set        */
 	int pt_enc;              /**< Payload type for encoding             */
-	int pt_dec;              /**< Payload type for decoding             */
 	bool rtcp_mux;           /**< RTP/RTCP multiplex supported by peer  */
 	bool jbuf_started;       /**< True if jitter-buffer was started     */
 	stream_pt_h *pth;        /**< Stream payload type handler           */

--- a/src/stream.c
+++ b/src/stream.c
@@ -293,12 +293,9 @@ static void rtp_handler(const struct sa *src, const struct rtp_header *hdr,
 	}
 
 	/* payload-type changed? */
-	if (s->pt_dec != hdr->pt) {
-		s->pt_dec = hdr->pt;
-		err = s->pth(hdr->pt, mb, s->arg);
-		if (err)
-			return;
-	}
+	err = s->pth(hdr->pt, mb, s->arg);
+	if (err)
+		return;
 
 	if (s->jbuf) {
 
@@ -526,7 +523,6 @@ int stream_alloc(struct stream **sp, struct list *streaml,
 	s->arg    = arg;
 	s->pseq   = -1;
 	s->ldir   = SDP_SENDRECV;
-	s->pt_dec = -1;
 
 	if (prm->use_rtp) {
 		err = stream_sock_alloc(s, prm->af);

